### PR TITLE
cc2DLabel added to the selected point when using the ccWaveformDialog

### PR DESCRIPTION
**Pull request by p-leroy**
It is practical to see where the selected point is when editing the index field.

### CODE REVIEW

File: qCC/ccWaveformDialog.cpp [hunk 1]
```diff
@@ -364,6 +364,8 @@ ccWaveDialog::ccWaveDialog(	ccPointCloud* cloud,
 	, m_pickingHub(pickingHub)
 	, m_gui(new Ui_WaveDialog)
 	, m_waveMax(0)
+	, m_label(std::shared_ptr<cc2DLabel>(new cc2DLabel()))
+	, m_display(cloud->getDisplay())
```
Comment (dgirardeau) [inline qCC/ccWaveformDialog.cpp:368]:
```suggestion
	, m_display(cloud ? cloud->getDisplay() : nullptr)
```

File: qCC/ccWaveformDialog.cpp [hunk 2]
```diff
@@ -397,15 +399,14 @@ ccWaveDialog::ccWaveDialog(	ccPointCloud* cloud,
...
-
 	//force update
 	onPointIndexChanged(0);
 }
 
 ccWaveDialog::~ccWaveDialog()
 {
-
 	delete m_gui;
+	m_display->redraw();
```
Comment (dgirardeau) [inline qCC/ccWaveformDialog.cpp:409]: test m_display

File: qCC/ccWaveformDialog.cpp [hunk 3]
```diff
@@ -422,9 +423,38 @@ void ccWaveDialog::onPointIndexChanged(int index)
...
+
+	add2DLabel(m_cloud, index);
+
 	m_widget->refresh();
 }
 
+bool ccWaveDialog::add2DLabel(ccPointCloud* cloud, unsigned pointIndex)
+{
+	ccGLCameraParameters camera;
+	m_display->getGLCameraParameters(camera);
```
Comment (dgirardeau) [inline qCC/ccWaveformDialog.cpp:435]: test m_display

File: qCC/ccWaveformDialog.cpp [hunk 4]
```diff
@@ -422,9 +423,38 @@ void ccWaveDialog::onPointIndexChanged(int index)
...
+
+	CCVector3 P = *cloud->getPoint(pointIndex);
```
Comment (dgirardeau) [inline qCC/ccWaveformDialog.cpp:437]:
```suggestion
	const CCVector3& P = *cloud->getPoint(pointIndex);
```

File: qCC/ccWaveformDialog.h [hunk 5]
```diff
@@ -144,6 +146,9 @@ class ccWaveDialog : public QDialog, public ccPickingListener
 
 	//! Maximum wave amplitude (for all points)
 	double m_waveMax;
+
+	std::shared_ptr<cc2DLabel> m_label;
```
Comment (dgirardeau) [inline qCC/ccWaveformDialog.h:150]: documentation

File: qCC/ccWaveformDialog.cpp (hunk 2 above)
Comment (dgirardeau) [reply qCC/ccWaveformDialog.cpp:409]: Where do you remove the label (if any)?

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: qCC/ccWaveformDialog.cpp (hunk 2 above)
Comment (p-leroy) [reply qCC/ccWaveformDialog.cpp:409]: m_label is an std::shared_ptr<cc2DLabel>, the label itself is removed when m_label is deleted, i.e. when the ccWaveformDialog is deleted

**Review by dgirardeau — DISMISSED** [review]
Thanks for the updates!

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**